apps/nextjs-website/src/components/organisms/GuideInPageMenu/components/Heading.tsx

```diff
@@ -0,0 +1,45 @@
+import { HeadingProps } from 'gitbook-docs/markdoc/schema/heading';
+import React, { ReactNode, useEffect, useState } from 'react';
+import MUILink from '@mui/material/Link';
+import { Typography, useTheme } from '@mui/material';
+import { useHash } from '@/components/organisms/HashProvider/HashProvider';
+
+const Heading = ({ level, id, children }: HeadingProps<ReactNode>) => {
+  const isString = typeof children === 'string';
+  const { palette } = useTheme();
+  const { hash } = useHash();
+  const [isCurrentHash, setIsCurrentHash] = useState(false);
+
+  useEffect(() => {
+    setIsCurrentHash(hash === `#${id}`);
+  }, [hash, id]);
+
+  if (level < 2 || level > 4) {
+    return;
+  }
```

Comment:
These two level (2,4) seems like magic number. If it not the case i suggest to add a comment to document this condition.